net/FlatBuffers/ByteBuffer.cs

```diff
@@ -42,19 +42,15 @@
 using System.Runtime.InteropServices;
 using System.Text;
 
-#if ENABLE_SPAN_T
+#if ENABLE_SPAN_T && NETSTANDARD2_1
```

Comment:
I think these checks need to >= NETSTANDARD2_1, since I assume it will work in all future versions.

Reply:
Unfortunately, C# doesn't allow for large and small comparisons to be made in the preprocessor.

[https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-if](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-if)


I can force a decision from a string by doing the following, but I feel it's too speculative to be used as a response to all future versions.

<DefineConstants Condition="$(TargetFramework.Contains('netstandard')) and $(TargetFramework.Length) > 12 and
    $([System.Version]::Parse('$(TargetFramework.Replace('netstandard',''))').CompareTo($([System.Version]::Parse('2.1')))) >= 0">
  NETSTANDARD2_1_OR_GREATER
</DefineConstants>


So my proposed approach is to write it in NETSTANDARD2_1 once.

This means that, for example, if Standard 5.0 exists and this project is referenced from Core 5.0,it will be treated as Standard 2.1(I created a Standard 2.0 project in my environment and referenced it from Core 3.1 to verify this)


If you need a reference from a newer framework in the future, you will need to create a NETSTANDARD2_1_OR_GREATER as follows.

With this approach, you only need to include the minimum version that you need, and that's all there is to it.

For example, assuming there is a future version 4.0, if the new features required are in 5.0 and will be referenced in the future by a 6.0 project, you can still get the behavior you want by writing.
(However, it felt redundant to use this method at the moment, so we went with the standard approach.)

<PropertyGroup Condition="'$(TargetFramework)'=='netstandard2.1' and '$(TargetFramework)'=='netstandard5.0'">
  <DefineConstants>NETSTANDARD2_1_OR_GREATER</DefineConstants>
</PropertyGroup>


Please let me know if there is a better way. I'll change it that way.

Reply:
I got this information from a follower of mine on twitter, but apparently Unity has a different definition of .NetStandard 2.0. So the root of the problem may be deeper than that.

[https://docs.unity3d.com/2018.4/Documentation/Manual/PlatformDependentCompilation.html](https://docs.unity3d.com/2018.4/Documentation/Manual/PlatformDependentCompilation.html)

I'll add support for Unity as soon as we have a policy in place.

Reply:
I was told this by a Twitter follower that .Net Standard doesn't seem to have a future.

Therefore, as I initially adopted, it seems to me that the 2.1 decision alone is fine for all future.

https://github.com/dotnet/designs/blob/main/accepted/2020/net5/net5.md

Reply:
Ok, good to know. I was always confused with the different flavors of .NET and glad to see some unification.